apps/ledger-live-desktop/src/renderer/components/DeviceAction/animations.ts

```diff
@@ -263,19 +283,20 @@ const blue = {
   },
 };
 
-const animations = { nanoX, nanoS, nanoSP, stax, blue };
-
-type InferredKeys = $Keys<typeof nanoS>;
+type Animations = {
+  [modelId in DeviceModelId]: DeviceAnimations;
+};
+const animations: Animations = { nanoX, nanoS, nanoSP, stax, blue };
 
 export const getDeviceAnimation = (
   modelId: DeviceModelId,
-  theme: "light" | "dark",
-  key: InferredKeys,
+  theme: Theme["theme"],
+  key: AnimationKey,
 ) => {
-  // $FlowFixMe Ignore the type to allow override from env.
-  modelId = process.env.OVERRIDE_MODEL_ID || modelId;
-  const lvl1 = animations[modelId] || animations.nanoX;
-  const lvl2 = lvl1[key] || animations.nanoX[key];
-  if (theme === "dark" && lvl2.dark) return lvl2.dark;
-  return lvl2.light;
+  const animationModelId = (process.env.OVERRIDE_MODEL_ID as DeviceModelId) || modelId;
+
+  // Handles the case where OVERRIDE_MODEL_ID is incorrect
+  const animation = animations[animationModelId][key][theme] || animations.nanoX[key][theme];
```

Comment:
You are ensuring that all models have a dark mode to avoid the 279 check, I wonder why we had such hacky code 🤔

Reply:
Yep - i think it's because no one wanted to change a code that was working